editoast/src/views/pagination.rs

```diff
@@ -65,16 +66,49 @@ fn default_page() -> i64 {
     1
 }
 
+impl PaginationQueryParam {
+    pub fn validate(&self, max_page_size: i64) -> Result<PaginationQueryParam> {
+        let page_size = self.page_size.unwrap_or(25).max(10);
+        if page_size > max_page_size || page_size < 1 || self.page < 1 {
+            return Err(PaginationError::InvalidPageSize {
+                provided_page_size: page_size,
+                max_page_size,
+            }
+            .into());
+        }
+        Ok(*self)
```

Comment:
Implicit copy here. Take the ownership `self` and don't dereference to avoid useless copies.